modules/snooker/src/test/ts/browser/ResizeTest.ts

```diff
@@ -191,29 +194,64 @@ UnitTest.test('ResizeTest', () => {
     const tableSize = TableSize.getTableSize(table);
 
     // 100% width table
-    assert.eq(800, tableSize.width());
+    assert.equal(800, tableSize.width());
 
     // 25% width delta
     const step = tableSize.getCellDelta(delta);
-    assert.eq(200, step);
+    assert.equal(200, step);
 
     const warehouse = Warehouse.fromTable(table);
     const widths = tableSize.getWidths(warehouse, tableSize);
 
     // [50%, 50%] existing widths.
-    assert.eq([ 400, 400 ], widths);
+    assert.deepEqual([ 398, 398 ], widths);
 
     const deltas = Deltas.determine(widths, 0, step, tableSize, resizing);
 
     // [25%, -25%] deltas.
-    assert.eq([ 200, -200 ], deltas);
+    assert.deepEqual([ 200, -200 ], deltas);
 
     Remove.remove(table);
-  };
+  });
+
+  it('TINY-7731: should handle resizing a table where a cell overflows it\'s specified size', () => {
+    const delta = 200;
+
+    const table = SugarElement.fromHtml<HTMLTableElement>(`<table style="border-collapse: collapse; width: 800px;">
+    <tbody>
+    <tr>
+    <td style="width: 400px;">thisisareallylongsentencewithoutspacesthatcausescontenttooverflow</td>
+    <td style="width: 400px;">B</td>
+    </tr>
+    <tr>
+    <td style="width: 400px;">C</td>
+    <td style="width: 400px;">D</td>
+    </tr>
+    </tbody>
+    </table>`);
 
-  percentTablePercentCellsTest();
-  percentTablePixelCellsTest();
-  pixelTablePixelCellsTest();
-  pixelTablePercentCellsTest();
-  Remove.remove(box);
+    Insert.append(box, table);
+
+    const tableSize = TableSize.getTableSize(table);
+
+    // 100% width table
+    assert.equal(800, tableSize.width());
+
+    // 25% width delta
+    const step = tableSize.getCellDelta(delta);
+    assert.equal(200, step);
+
+    const warehouse = Warehouse.fromTable(table);
+    const widths = tableSize.getWidths(warehouse, tableSize);
+
+    assert.approximately(widths[0], 483, 1, 'First column width');
```

Comment:
Can you put a comment explaining these numbers? Even if it's just something like
```suggestion
    // Calculated experimentally as the width of "thisisareallylongsentencewithoutspacesthatcausescontenttooverflow"
    assert.approximately(widths[0], 483, 1, 'First column width');
```

(assuming that's where the numbers come from)

Reply:
```suggestion

    // This is the width of "thisisareallylongsentencewithoutspacesthatcausescontenttooverflow" which can vary marginally between browsers
    assert.approximately(widths[0], 483, 1, 'First column width');
```